### PR TITLE
reconcile lockdown, harden, purify, dump, replay

### PR DESCRIPTION
**Pull request by kriskowal**
Commits are individually reviewable. This subsumes the stack:

- https://github.com/agoric-labs/xsnap-pub/pull/8
- https://github.com/agoric-labs/xsnap-pub/pull/5
- https://github.com/agoric-labs/xsnap-pub/pull/9
- https://github.com/agoric-labs/xsnap-pub/pull/10
- https://github.com/agoric-labs/xsnap-pub/pull/6
- https://github.com/agoric-labs/xsnap-pub/pull/11
- https://github.com/agoric-labs/xsnap-pub/pull/12

### CODE REVIEW

**Comment by dckc** [comment]
The net of this clean-up is:

https://github.com/agoric-labs/xsnap-pub/compare/6a29ad9..4994e6e

File: xsnap/sources/xsnap.c [hunk 1]
```diff
@@ -6,13 +6,13 @@ static void xsBuildAgent(xsMachine* the);
 static void xsPlayTest(xsMachine* the);
 static void xsPrintUsage();
 
-static void xs_clearTimer(xsMachine* the);
+//static void xs_clearTimer(xsMachine* the);
```
Comment (dckc) [inline xsnap/sources/xsnap.c:11]: @kriskowal maybe we don't want to turn these timer functions off after all.

File: xsnap/sources/xsnap.c [hunk 2]
```diff
@@ -294,9 +324,8 @@ void xsBuildAgent(xsMachine* machine)
 	xsResult = xsNewHostFunction(xs_issueCommand, 1);
 	xsDefine(xsGlobal, xsID("issueCommand"), xsResult, xsDontEnum);
 	
-	xsVar(0) = xsGet(xsGlobal, xsID("Date"));
-	xsVar(0) = xsGet(xsVar(0), xsID("now"));
```
Comment (dckc) [inline xsnap/sources/xsnap.c:288]: this looks like a git-oopsie.

@kriskowal to take a look.

**Comment by dckc** [comment]
net change from master https://github.com/agoric-labs/xsnap-pub/compare/57dcfc0..4994e6e

File: xsnap/sources/xsnap.c (hunk 2 above)
Comment (kriskowal) [reply xsnap/sources/xsnap.c:288]: I haven’t traced the diff confusion, but the end state is that both xsnap.c and xsnap-worker.c use the same high resolution timer native and that this `performance.now = Date.now` stop-gap has been converted.

File: xsnap/sources/xsnap.c (hunk 1 above)
Comment (kriskowal) [reply xsnap/sources/xsnap.c]: Removed and adjusted the callback tables all they way from here to the HEAD of the stack.

**Comment by kriskowal** [comment]
Resolved findings above and send to validate in Agoric CI now: https://github.com/Agoric/agoric-sdk/pull/3912/commits/4650bb4e6f90aeca5ed8b62240b49aa55524184f

**Review by phoddie — APPROVED** [review]
- xsnap.c looks good
- xsnap-worker.c looks OK, to the extend I understand. There are some differences between xsnap.c and xsnap-worker.c that don't seem necessary - the closer the sources of those two files can be, the easier it will be to migrate changes
- xsnapPlatform.c (and most everything else) are unchanged from Moddable's development version

**Review by dckc — APPROVED** [review]

**Comment by dckc** [comment]
> the closer the sources of those two files can be, the easier it will be to migrate changes

Agreed. @kriskowal and I brainstormed a bit about combining them, in fact.

But we can address remaining differences after landing this PR.

over to you, @kriskowal

**Pull request merged**